packages/dynamite/dynamite_runtime/lib/src/client/client.dart

```diff
@@ -97,8 +95,28 @@ class DynamiteClient {
       request.bodyBytes = body;
     }
 
+    final response = await send(request);
+
+    if (validStatuses?.contains(response.statusCode) ?? true) {
+      return response;
+    } else {
+      if (kDebugMode) {
+        final result = await http.Response.fromStream(response);
+        throw DynamiteStatusCodeException.fromResponse(result);
+      } else {
+        throw DynamiteStatusCodeException(response.statusCode);
+      }
+    }
+  }
+
+  @override
+  Future<http.StreamedResponse> send(http.BaseRequest request) async {
+    baseHeaders?.forEach((key, value) {
+      request.headers.putIfAbsent(key, () => value);
```

Comment:
Shouldn't the base headers always override the request headers? At least that is how it worked so far. I'm not sure which one is usually the expected behavior.

Reply:
1. so far the request headers did overwrite the base ones as we added them later on, so no change here:
```dart

    if (baseHeaders != null) {
      request.headers.addAll(baseHeaders!);
    }

    if (headers != null) {
      request.headers.addAll(headers);
    }
```
> If a key of other is already in this map, its value is overwritten.

https://api.dart.dev/stable/3.3.0/dart-core/Map/addAll.html

2. I think that this is the right thing to do as a user might add arbitrary headers to the baseClient that are not anticipated by openapi spec potentially leading in invalid requests.
Therefore a request should always set the headers it requires (I think we did already discuss this once :thinking:)

Reply:
Ok makes sense and yeah we probably had this discussion already :D
Maybe you can add a comment explaining it, so I won't ask again next time.